tests/unit/test_pretty.py

```diff
@@ -150,3 +150,28 @@ def test_large():
     )
     assert str(status) == status_repr
     assert eval(status_repr, jubilant.statustypes.__dict__) == status
+
+
+def test_diff():
+    # It's simplest to test _status_diff directly, even though it's not public.
+    old_json = json.loads(DATABASE_WEBAPP_JSON)
+    new_json = json.loads(DATABASE_WEBAPP_JSON)
+    new_json['applications']['database']['application-status']['current'] = 'waiting'
+    new_json['applications']['database']['relations']['db'][0]['scope'] = 'testy'
+    del new_json['applications']['database']['relations']['db'][1]
+    old_status = jubilant.Status._from_dict(old_json)
+    new_status = jubilant.Status._from_dict(new_json)
+
+    diff = jubilant._juju._status_diff(old_status, new_status)
+
+    assert (
+        diff
+        == """\
```

Comment:
nitpick: I'm allergic to line continuation characters, so I'd personally do this instead:

```
assert foo == """
text
text
""".strip()
```

Reply:
Some folks like to indent such test vectors and dedent on the fly.

Others move the expected value to a module-level constant to make the test look nicer.

I don't particularly care, just shooting ideas 🙇🏻

Reply:
That's not bad, though I prefer the direct approach with the `\` (James suggested it originally) rather than dedent or strip, so I'm going to leave this as is. It's also consistent with the other such strings above.